docs/source/conf.py

```diff
@@ -149,6 +149,9 @@
     ('py:class', 'firedrake.solving_utils._SNESContext'),
     # Pyadjoint BlockVariable is not properly referenced in pyadjoint
     ('py:class', 'BlockVariable'),
+    # Cofunction.ufl_domains references FormArgument but it isn't picked
+    # up by Sphinx (see https://github.com/sphinx-doc/sphinx/issues/11225)
+    ('py:class', 'FormArgument'),
```

Comment:
Please can you open a UFL issue for this, and a corresponding Firedrake issue to hopefully remove this once it's fixed?